### PR TITLE
gh-86986: ﻿﻿Drop compatibility support for Sphinx 2

### PR DESCRIPTION
**Pull request by AA-Turner**
Reverts two earlier commits and removes Python 2 support from `suspicious.py`.

A

### CODE REVIEW

**Review by ezio-melotti — COMMENTED** [review]
The PR itself LGTM, but was there some discussion somewhere about this?
For context, the Sphinx version was bumped to 3.2 in Python 3.10+ in https://github.com/python/cpython/issues/86986 and https://github.com/python/cpython/pull/93337, so it should be fine to apply this to `main` and backport it to `3.11` and `3.10` too.

**Comment by AA-Turner** [comment]
> was there some discussion somewhere about this?

I can open an issue if you'd like? This started out as changes as part of #93738, but I realised that they could stand alone so split them out.

A

**Comment by hugovk** [comment]
There was discussion in May 2021 about needing Sphinx 1.8 compatibility: https://github.com/python/python-docs-theme/pull/44#issuecomment-837965838

Here's some discussion in May 2022 about requiring Sphinx 3.2 as a minimum: https://github.com/python/cpython/pull/92318#issuecomment-1120482146 (with no objections) which led to https://github.com/python/cpython/pull/93337 bumping to 3.2+ this month.

File: Doc/conf.py [hunk 1]
```diff
@@ -237,5 +237,3 @@
 # bpo-40204: Disable warnings on Sphinx 2 syntax of the C domain since the
 # documentation is built with -W (warnings treated as errors).
 c_warn_on_allowed_pre_v3 = False
```
Comment (hugovk) [inline Doc/conf.py:239]: Also part of this "Sphinx 2 and Sphinx 3 compatibility" section:

Are `c_allow_pre_v3` and `c_warn_on_allowed_pre_v3` above covered by https://github.com/python/cpython/issues/93738? Or can they removed here?

**Review by hugovk — APPROVED** [review]

File: Doc/conf.py (hunk 1 above)
Comment (AA-Turner) [reply Doc/conf.py:239]: If you remove it you get around 1,300 lines of warnings and then a build failure... Hence #93738! Yeah, they need to be kept here.

A

**Review by ezio-melotti — APPROVED** [review]

**Pull request merged**